File: mifosng-android/src/main/java/com/mifos/api/services/SearchService.java
/*
 * This project is licensed under the open source MPL V2.
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */
package com.mifos.api.services;

import com.mifos.api.model.APIEndPoint;
import com.mifos.objects.SearchedEntity;

import java.util.List;

import retrofit2.http.GET;
import retrofit2.http.Query;
import rx.Observable;

/**
 * @author fomenkoo
 */
public interface SearchService {

    @GET(APIEndPoint.SEARCH)
    Observable<List<SearchedEntity>> searchResources(@Query("query") String clientName,
                                                     @Query("resource") String resources,
                                                     @Query("exactMatch") Boolean exactMatch);
}

File: mifosng-android/src/main/java/com/mifos/api/services/SearchService.kt
/*
 * This project is licensed under the open source MPL V2.
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */
package com.mifos.api.services

import com.mifos.api.model.APIEndPoint
import com.mifos.objects.SearchedEntity
import retrofit2.http.GET
import retrofit2.http.Query
import rx.Observable

/**
 * @author fomenkoo
 */
interface SearchService {
    @GET(APIEndPoint.SEARCH)
    suspend fun searchResources(
        @Query("query") clientName: String?,
        @Query("resource") resources: String?,
        @Query("exactMatch") exactMatch: Boolean?
    ): List<SearchedEntity>
}